File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/cpp/exercises/practice/grade-school/grade_school.cpp
#include "grade_school.h"

namespace grade_school {

bool school::add(const std::string& name, int grade) {
    // Check all grades for existing student
    for (const auto& [g, students] : roster_) {
        if (students.count(name)) {
            return false;
        }
    }
    
    roster_[grade].insert(name);
    return true;
}

std::vector<std::string> school::roster() const {
    std::vector<std::string> result;
    for (const auto& [grade, students] : roster_) {
        result.insert(result.end(), students.begin(), students.end());
    }
    return result;
}

std::set<std::string> school::grade(int grade) const {
    if (auto it = roster_.find(grade); it != roster_.end()) {
        return it->second;
    }
    return {};
}

}  // namespace grade_school

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/javascript/exercises/practice/grade-school/grade-school.js
//
// This is only a SKELETON file for the 'Grade School' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class GradeSchool {
  constructor() {
    this._students = new Map();
  }

  roster() {
    const sorted = {};
    // Get sorted grades and sort each grade's names
    Array.from(this._students.keys())
      .sort((a, b) => a - b)
      .forEach(grade => {
        sorted[grade] = this.grade(grade);
      });
    return sorted;
  }

  add(name, grade) {
    // Check if student exists and remove from current grade
    const currentGrade = this._findGrade(name);
    if (currentGrade !== undefined) {
      this._students.get(currentGrade).delete(name);
    }
    
    // Add to new grade
    if (!this._students.has(grade)) {
      this._students.set(grade, new Set());
    }
    this._students.get(grade).add(name);
  }

  grade(gradeNumber) {
    const grade = this._students.get(gradeNumber);
    return grade ? Array.from(grade).sort() : [];
  }

  _findGrade(name) {
    // Find which grade the student is in
    for (const [grade, students] of this._students.entries()) {
      if (students.has(name)) {
        return grade;
      }
    }
    return undefined;
  }
}
